Issues with latest SharpGenTools (ci446)

When switching to SharpGenTools `1.1.1-ci446` , there are lots of compilation regressions related to unsafe context not set or to some duplicated local variables:

![image](https://user-images.githubusercontent.com/715038/42622508-d592a3d2-85c0-11e8-82cb-2f73696ae7dd.png)

When trying to compile this branch on my machine. I'm getting the following errors:

```
error MSB4018: The "GenerateCSharp" task failed unexpectedly.
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018: System.InvalidOperationException: No marshaller found for SharpDX.DirectWrite.GdiInterop1.GetFontSignature.fontSignature
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.MarshallingRegistry.GetMarshaller(CsMarshalBase csElement)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.CallableCodeGenerator.<GenerateCode>d__5.MoveNext()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.MethodCodeGenerator.<GenerateCode>d__4.MoveNext()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at System.Collections.Generic.List`1.AddRange(IEnumerable`1 collection)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.InterfaceCodeGenerator.<GenerateCode>d__5.MoveNext()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at Microsoft.CodeAnalysis.CSharp.SyntaxFactory.List[TNode](IEnumerable`1 nodes)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.RoslynGenerator.GenerateCompilationUnit[T](String csNamespace, IEnumerable`1 elements, IMultiCodeGenerator`2 generator)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGen.Generator.RoslynGenerator.Run(CsSolution solution, String rootDirectory, String generatedCodeFolder, Boolean includeAssemblyNameFolder)
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at SharpGenTools.Sdk.Tasks.GenerateCSharp.Execute()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
1>C:\Users\xoofx\.nuget\packages\sharpgentools.sdk\1.1.1-ci458\build\SharpGenTools.Sdk.targets(220,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
1>Done building project "SharpDX.csproj" -- FAILED.
```

--- a/Source/Directory.build.props
+++ b/Source/Directory.build.props
@@ -50,7 +50,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci446" PrivateAssets="All" />
+    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci458" PrivateAssets="All" />
     <PackageReference Include="SharpGen.Doc.Msdn.Tasks" Version="1.1.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
     <SharpGenGlobalNamespaceOverrides Include="InterfaceArray" Override="ComArray" />

--- a/Source/Directory.build.props
+++ b/Source/Directory.build.props
@@ -50,7 +50,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci458" PrivateAssets="All" />
+    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci461" PrivateAssets="All" />
     <PackageReference Include="SharpGen.Doc.Msdn.Tasks" Version="1.1.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
     <SharpGenGlobalNamespaceOverrides Include="InterfaceArray" Override="ComArray" />

--- a/Source/Directory.build.props
+++ b/Source/Directory.build.props
@@ -50,7 +50,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.0" PrivateAssets="All" />
+    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci446" PrivateAssets="All" />
     <PackageReference Include="SharpGen.Doc.Msdn.Tasks" Version="1.1.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
     <SharpGenGlobalNamespaceOverrides Include="InterfaceArray" Override="ComArray" />

--- a/Source/Directory.build.props
+++ b/Source/Directory.build.props
@@ -50,7 +50,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci461" PrivateAssets="All" />
+    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci458" PrivateAssets="All" />
     <PackageReference Include="SharpGen.Doc.Msdn.Tasks" Version="1.1.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
     <SharpGenGlobalNamespaceOverrides Include="InterfaceArray" Override="ComArray" />

--- a/Source/Directory.build.props
+++ b/Source/Directory.build.props
@@ -50,7 +50,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci458" PrivateAssets="All" />
+    <PackageReference Include="SharpGenTools.Sdk" Version="1.1.1-ci462" PrivateAssets="All" />
     <PackageReference Include="SharpGen.Doc.Msdn.Tasks" Version="1.1.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
     <SharpGenGlobalNamespaceOverrides Include="InterfaceArray" Override="ComArray" />

--- a/Source/SharpDX.Direct2D1/Mapping-direct2d1.xml
+++ b/Source/SharpDX.Direct2D1/Mapping-direct2d1.xml
@@ -86,6 +86,8 @@
 
     <create class="D2D1" />
     <context-clear />
+
+    <define struct="SharpDX.DirectWrite.GdiInterop.FontSignature" sizeof="24" />
   </extension>
   
   <bindings>

--- a/Source/SharpDX.Direct2D1/Mapping-direct2d1.xml
+++ b/Source/SharpDX.Direct2D1/Mapping-direct2d1.xml
@@ -65,7 +65,7 @@
     <context>sharpdx-direct2d1-ext</context>
     <context>dcommon</context>
     <context>d2d1</context>
-    <context>d2derr</context>
+    <context>D2DErr</context>
     <context>d2d1_1</context>
     <context>d2d1_2</context>
     <context>d2d1_3</context>

--- a/Source/SharpDX.Direct3D12/ShaderBytecode.cs
+++ b/Source/SharpDX.Direct3D12/ShaderBytecode.cs
@@ -73,7 +73,7 @@ namespace SharpDX.Direct3D12
             native.Pointer = pinBuffer;
             if (managedData != null)
             {
-                native.Size = managedData.Length;
+                native.Size = (IntPtr)managedData.Length;
             }
         }
     }

